### PR TITLE
Unstake Input fix

### LINKED ISSUE #376: Invalid input value for unstake command
#### Describe the bug
Input value that staker wants to unstake is considered  as number of RZR's and same is being sent for unstake transaction to contracts. While in contracts, unstake function expects number of sRZR's to be passed as parameter but razor-go passes number of RZR's as a parameter

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

The input value for unstake is considered in RZR's and than it is converted into its equivalent sRZR's that is passed as parameter to unstake contract call.

Fixes #376

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cmd/unstake.go [hunk 1]
```diff
@@ -99,12 +100,48 @@ func Unstake(config types.Configurations, client *ethclient.Client, input types.
...
 		return txnArgs, err
 	}
 
+	staker, err := utilsStruct.razorUtils.GetStaker(client, txnArgs.AccountAddress, stakerId)
+	if err != nil {
+		log.Error("Error in getting staker: ", err)
+		return txnArgs, err
+	}
+
+	stakedToken := utilsStruct.razorUtils.GetStakedToken(client, staker.TokenAddress)
```
Comment (ashish10677) [inline cmd/unstake.go:116]: Can you move this entire process of getting balance of sRzr to a new function.

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: cmd/unstake.go (hunk 1 above)
Comment (Yashk767) [reply cmd/unstake.go:116]: yes

File: cmd/unstake.go [hunk 2]
```diff
@@ -177,6 +154,38 @@ func AutoWithdraw(txnArgs types.TransactionOptions, stakerId uint32, utilsStruct
...
+	}
+
+	totalSupply, err := utilsStruct.stakeManagerUtils.GetTotalSupply(stakedToken, &callOpts)
+	if err != nil {
+		log.Error("Error in getting total supply: ", err)
+		return nil, err
+	}
+
+	maxUnstake := utilsStruct.razorUtils.ConvertSRZRToRZR(sRZRBalance, staker.Stake, totalSupply)
+	log.Infof("The maximum RZRs you can unstake: %f RZRs", utilsStruct.razorUtils.GetAmountInDecimal(maxUnstake))
```
Comment (ashish10677) [inline cmd/unstake.go]: Make this log.Debugf

Comment (Yashk767) [reply cmd/unstake.go]: okay

File: cmd/unstake_test.go [hunk 3]
```diff
@@ -432,3 +474,154 @@ func TestAutoWithdraw(t *testing.T) {
...
+				totalSupply:   big.NewInt(1000),
+				RZR:           big.NewInt(1000),
+				decimalAmount: big.NewFloat(1000),
+				sRZR:          big.NewInt(1000),
+			},
+			want:    nil,
+			wantErr: errors.New("invalid amount"),
+		},
+		{
+			name: "Test 5: When there is an error in converting RZR's to sRZR's",
```
Comment (ashish10677) [inline cmd/unstake_test.go:565]: Add one more test case with real world values, where the supply is 1 billion razors.

File: utils/math_test.go [hunk 4]
```diff
@@ -794,3 +794,99 @@ func Test_getFractionalWeight(t *testing.T) {
...
+			name: "Test 3: When currentStake is 0",
+			args: args{
+				amount:       big.NewInt(500),
+				currentStake: big.NewInt(0),
+				totalSupply:  big.NewInt(2000),
+			},
+			want:    big.NewInt(0),
+			wantErr: true,
+		},
+	}
```
Comment (ashish10677) [inline utils/math_test.go:840]: Add tests here with higher values as well.

File: utils/math_test.go [hunk 5]
```diff
@@ -794,3 +794,99 @@ func Test_getFractionalWeight(t *testing.T) {
...
+		},
+		{
+			name: "Test 2: When totalSupply < currentStake",
+			args: args{
+				sAmount:      big.NewInt(500),
+				currentStake: big.NewInt(2000),
+				totalSupply:  big.NewInt(1000),
+			},
+			want: big.NewInt(1000),
+		},
```
Comment (ashish10677) [inline utils/math_test.go:883]: Same thing here.

**Review by ashish10677 — CHANGES_REQUESTED** [review]

**Review by ashish10677 — APPROVED** [review]

**Pull request merged**